### ISSUE TITLE
Port storage to db_ledger

### ISSUE THREAD

**Issue by mvines** [solana-labs/solana#2308]
#2298 broke storage as the old ledger is no longer written to disk, and I punted on porting storage to db_ledger in that effort.  Here's roughly what needs to be done:

* [x] Fill out `db_ledger:: get_entries_bytes`, based on `ledger::LedgerWindow::get_entries_bytes` to make `chacha_cbc_encrypt_file_many_keys()` work
* [x] Update `src/replicator.rs` to use db_ledger
* [x] Make `ledger::LEDGER_DATA_FILE` private to ensure all users outside of ledger.rs have been purged
* [x] Fix test_encrypt_file_many_keys_multiple_keys
* [x] Restore this test: `test_encrypt_file_many_keys_bad_key_length`

**Comment by sakridge** [comment]
Well, I wonder if the best way is to re-use ledger but store serialized blobs per line instead and make the window service configurable on storing to db_ledger or the new blob_ledger. @rob-solana what do you think?

**Comment by rob-solana** [comment]
@sakridge that might be a workable stopgap, get your work unblocked.  I'd prefer that db_ledger be the thing that supports your APIs, though, so we can get a full picture of the requirements of the blob store

**Comment by sakridge** [comment]
Ok, yea I don't think we should spend much time on a stop-gap, I say go with db_ledger if that's what we want to store the necessary blob metadata needed for storage.

**Comment by rob-solana** [comment]
that was fast ;)

**Comment by sakridge** [comment]
@rob-solana Well not entirely done yet. I split it into another issue #2338 with some more work, I just wanted to get one PR merged which handles the list of stuff in this description.
